### ISSUE TITLE
Delta row tree data doesn't respect row order

### ISSUE THREAD

**Issue by Abrissirba** [ag-grid/ag-grid#2339]
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
I'm having a tree with deltaRowDataMode set to true where the rowData array comes from a redux store. When you add new items to the array they are always added at the end. It doesn't matter at what position the item is added.

**Expected behavior**
If an item is added first to the data array it should also be first in the tree, given that no sorting is applied. It seems to be working for a flat list but not for tree data, (example of adding first to a flat list https://plnkr.co/edit/?p=preview)

**Minimal reproduction of the problem with instructions**
I used the Redux file explorer example and changed so that the new file is added first in the files array in fileReducer.jsx. When adding a new file it is added to the end in the grid even though it is added first to the data array.

https://plnkr.co/edit/xDUrBRMrjUc4300IA8F1?p=preview

**What is the motivation / use case for changing the behavior?**
Being able to use ag-grid with a redux store

* **ag-Grid version:** X.X.X
17.1.0

* **Browser:**
Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript

**Comment by Abrissirba** [comment]
This issue is now tracked as a bug with key AG-1749 at https://www.ag-grid.com/ag-grid-pipeline/

As a workaround until it'is fixed I'm doing the following

    this.gridOptions.deltaRowDataMode = false;
    this.gridApi.setRowData(response);
    this.gridOptions.deltaRowDataMode = true;

**Comment by makinggoodsoftware** [comment]
Hi,

I can confirm that this was fixed in v18.0.0. I am going to close this issue

Thanks
